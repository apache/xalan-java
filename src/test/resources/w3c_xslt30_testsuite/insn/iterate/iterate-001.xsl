<xsl:stylesheet version="3.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">

  <!-- A primitive test for the XSLT xsl:iterate instruction. Exercises 
       position() and last() functions. -->

  <xsl:output method="xml" indent="yes"/>
  
  <xsl:template match="/">
     <out>
        <xsl:iterate select="//ITEM/TITLE">
           <item position="{position()}" last="{last()}">
               <xsl:copy-of select="."/>
           </item>
        </xsl:iterate>
     </out>
  </xsl:template>

</xsl:stylesheet>
