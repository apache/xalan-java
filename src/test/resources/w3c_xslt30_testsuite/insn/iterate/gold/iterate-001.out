<?xml version="1.0" encoding="UTF-8"?><out>
  <item last="6" position="1">
    <TITLE>Pride and Prejudice</TITLE>
  </item>
  <item last="6" position="2">
    <TITLE>Wuthering Heights</TITLE>
  </item>
  <item last="6" position="3">
    <TITLE>Tess of the d'Urbervilles</TITLE>
  </item>
  <item last="6" position="4">
    <TITLE>Jude the Obscure</TITLE>
  </item>
  <item last="6" position="5">
    <TITLE>The Big Over Easy</TITLE>
  </item>
  <item last="6" position="6">
    <TITLE>The Eyre Affair</TITLE>
  </item>
</out>
